ui/desktop/src/components/ChatInput.tsx

```diff
@@ -302,15 +302,19 @@ export default function ChatInput({
 
   // Handle recipe prompt updates
   useEffect(() => {
-    // If recipe is accepted and we have an initial prompt, and no messages yet, and we haven't set it before
+    const requiresParams = (recipeConfig?.parameters?.length ?? 0) > 0;
     if (recipeAccepted && initialPrompt && messages.length === 0) {
-      setDisplayValue(initialPrompt);
-      setValue(initialPrompt);
-      setTimeout(() => {
-        textAreaRef.current?.focus();
-      }, 0);
+      // If recipe requires parameters, only proceed if we have them or if initialPrompt is already substituted
+      // (initialPrompt will be substituted after parameters are collected)
+      if (!requiresParams || !initialPrompt.includes('{{')) {
+        setDisplayValue(initialPrompt);
```

Comment:
does this mean that if you have a recipe whose initial prompt contains {{ but nothing else, it will never show? this feels like it should be done with two values, or make it so that initialPrompt only has a value after we resolve the variables